crates/primitives/starknet/src/transaction/mod.rs

```diff
@@ -40,19 +45,60 @@ impl Event {
         Self {
             keys: BoundedVec::try_from(vec![]).unwrap(),
             data: BoundedVec::try_from(vec![]).unwrap(),
-            from_address: H256::zero(),
+            from_address: ContractAddressWrapper::default(),
+        }
+    }
+
+    /// Set from address.
+    pub fn set_from_address(&self, from_address: StarknetContractAddress) -> Self {
+        Self {
+            keys: self.keys.clone(),
+            data: self.data.clone(),
+            from_address: from_address.0.key().bytes().try_into().unwrap(),
         }
     }
 }
-impl Default for Event {
+impl Default for EventWrapper {
     fn default() -> Self {
         let one = H256::from_slice(&[
             0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1,
         ]);
         Self {
             keys: BoundedVec::try_from(vec![one, one]).unwrap(),
             data: BoundedVec::try_from(vec![one, one]).unwrap(),
-            from_address: one,
+            from_address: ContractAddressWrapper::from(one),
+        }
+    }
+}
+
+impl From<Event> for EventWrapper {
+    fn from(event: Event) -> Self {
+        Self {
+            keys: BoundedVec::try_from(
+                event.content.keys.iter().map(|k| H256::from_slice(k.0.bytes())).collect::<vec::Vec<H256>>(),
+            )
+            .unwrap(),
+            data: BoundedVec::try_from(
+                event.content.data.0.iter().map(|d| H256::from_slice(d.bytes())).collect::<vec::Vec<H256>>(),
+            )
+            .unwrap(),
+            from_address: event.from_address.0.key().bytes().try_into().unwrap(),
+        }
+    }
+}
+
+impl From<EventContent> for EventWrapper {
+    fn from(event: EventContent) -> Self {
+        Self {
+            keys: BoundedVec::try_from(
+                event.keys.iter().map(|k| H256::from_slice(k.0.bytes())).collect::<vec::Vec<H256>>(),
+            )
+            .unwrap(),
```

Comment:
since it can fail it should be a try_from and return a result. we should avoid using unwrap